src/lib.rs

```diff
@@ -1117,14 +1130,27 @@ impl<T: Counter> Histogram<T> {
 
     /// Computes the matching histogram value for the given histogram bin.
     pub fn value_for(&self, index: usize) -> u64 {
+        // TODO what to do about indexes that are beyond what can be represented by this histogram?
+        // Pretty easy to end up shifting off the high end of a u64, e.g. asking for a big index
+        // when sigfigs is high.
```

Comment:
It was getting complicated (for my brain, not the computer) to juggle multiple PRs so I picked a few small simplifications to include here since they fit well with the theme of closer type bounds.